File: codes/c++/PartitionArrayForMaximumSum.cpp
#include <iostream>

#include <bits/stdc++.h>

using namespace std;

int main(){
    vector<int> arr={15,7,9,2,5,10,3}; int k = 3;
    //[15,15,15,9,10,10,10]
    //cout <<arr.size()<<endl;
  
    sort(arr.begin(), arr.end(), greater<int>());
    //sort(arr.begin(), arr.end());
    cout <<"After sorting: ";
    for(auto x: arr){
        cout << x << " ";
    }
    cout << endl;
    
    vector<int> maxValues;
 
    for(int i=0; i<arr.size(); ++i){
        if((arr[i]>arr[i+1]) && (arr[i]!=arr[i+1])){
            if(k==0){
                break;
            }
            if(k!=0){
                k=k-1;
                maxValues.push_back(arr[i]);
                 
            }
           
        }
    }



    cout<<"maxValues ";
    for(auto x: maxValues){
        cout << x << " ";
    }
    cout <<endl;
    int result=0;
    
    int count=0;
    cout <<maxValues[0]<<endl;

    int mod = arr.size()%k;
    cout <<"MOD "<<mod<<endl;
    int div = arr.size()/k;
    cout <<"run "<<mod<<endl;

    return 0;
}

File: codes/js/partition-array-for-maximum-sum.js
console.log("Hello, World!");


    
//Given
const arr=[1,15,7,9,2,5,10]; let k=3;

//baseCase
if(arr.length==1 && k==1){
        return arr[0];
    }

//sorting array in descending order
var arrayLength = arr.length;
arr.sort(function(a, b){return b - a});

/*
//print arrray in same line

var string = "";
  arr.forEach(function(element){
      string += element;
  });
console.log(string); //123
*/

//checking the sorted array
for (var i = 0; i < arrayLength; i++) {
console.log(arr[i]);
//Do something
}

let div = arrayLength/k;
div=Math.round(div);
console.log(div);
let mod = arrayLength%k;
console.log(mod);
let result=0;
let iCounter=0;
//for div
for(var i=0;i<div;++i){
  
  for(var j =0;j<k;++j){
    result+=arr[i]
  }
  
  ++iCounter;
}


console.log(result)

for(var i=0;i<mod;++i){
  
  for(var j =0;j<(k-div);++j){
    result+=arr[iCounter]
  }
  
  ++iCounter;
}
console.log(result)
/*let remain=arrayLength-div;
if(mod!=0){
  for(var i=0;)
}*/